src/machines/workflow_task_state_machine.rs

```diff
@@ -99,7 +106,29 @@ impl TryFrom<HistoryEvent> for WorkflowTaskMachineEvents {
             }),
             Some(EventType::WorkflowTaskTimedOut) => Self::WorkflowTaskTimedOut,
             Some(EventType::WorkflowTaskCompleted) => Self::WorkflowTaskCompleted,
-            Some(EventType::WorkflowTaskFailed) => Self::WorkflowTaskFailed,
+            Some(EventType::WorkflowTaskFailed) => Self::WorkflowTaskFailed(WFTFailedDat {
+                // TODO(maxim): How to avoid clone of attributes. We need to borrow e for the
+                // MalformedEvent down there. But forcing clone just for that doesn't make sense.
```

Comment:
The key here is that malformed event doesn't borrow `e`, it consumes it, so you somehow need to tell the compiler "if this struct has the data I need, then consume it and take `new_run_id` out, but if it doesn't, then give it to `MalformedEvent` which will also consume it."

Actually doing that exact thing is probably not worth the effort over just cloning, but it doesn't need to be all of the attributes. You only care about the run id, so the thing to do here would be to attempt to extract that first, clone it, and if it didn't work feed `e` to `MalformedEvent`. That's similar to what I do above with `WorkflowTaskStarted`. Cloning a run id is very cheap in the grand scheme of things compared to complicating the code to avoid it -- but you might want to give it a shot anyway. Puzzles like this are a great way to learn Rust. I will say that Rust often puts you in this mindset of trying to never clone things or `Box` them or whatever because it's super fun to play the optimization game, but you have to remember Java is cloning and `Box`ing things *all over the place*, so it's often worth stepping back and remembering it's probably not worth optimizing prematurely.